jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpField.java

```diff
@@ -362,6 +362,56 @@ else if (c != '0' && c != '.')
         return param != ZERO_QUALITY.length() && match == search.length();
     }
 
+    /**
+     * Look for a value as the last value in a possible multivalued field
+     * Parameters and specifically quality parameters are not considered.
+     * @param search Values to search for (case-insensitive)
+     * @return True iff the value is contained in the field value entirely or
+     * as the last element of a quoted comma separated list.
+     */
+    public boolean containsLast(String search)
+    {
+        return containsLast(getValue(), search);
+    }
+
+    /**
+     * Look for the last value in a possible multivalued field
+     * Parameters and specifically quality parameters are not considered.
+     * @param value The field value to search in.
+     * @param search Values to search for (case-insensitive)
+     * @return True iff the value is contained in the field value entirely or
+     * as the last element of a quoted comma separated list.
+     */
+    public static boolean containsLast(String value, String search)
+    {
+        if (search == null)
+            return value == null;
+        if (search.isEmpty())
+            return false;
+        if (value == null)
+            return false;
+        if (search.equalsIgnoreCase(value))
+            return true;
+
+        if (value.endsWith(search))
```

Comment:
Could we do this in a better way? Here we always do an `endsWith`, then we possibly just create a `QuotedCSV` and a `equalsIgnoreCase` anyway.

Reply:
Creating a QuotedCSV is very expensive compared to a quick endsWith check.    Very few actual headers will be lists... vanishingly few.  We only have the QuotedCSV path for correctness